### PR TITLE
Chat

### PR DESCRIPTION
**Pull request by mstopa**

### CODE REVIEW

File: tp_models/users/controller.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def add_user():
         return e.message, e.status
 
     token = generate_confirmation_token(email)
-    send_email(email, f"Confirmation Email for {username}", f"Click here to confirm! -> http://127.0.0.1:5000/confirm/{token}")
+    send_email(email, f"Confirmation Email for {username}", f"Click here to confirm! -> http://127.0.0.1:5000/users/confirm/{token}")
```
Comment (cansubmarinesswim) [inline tp_models/users/controller.py]: Is there a reason not to include token generation and sending email confirmation within the `try` block? And handle appropriate exceptions that could occur (possibly from smtplib)

File: setup.py [hunk 2]
```diff
@@ -3,9 +3,8 @@
 from distutils.core import setup
```
Comment (cansubmarinesswim) [inline setup.py:3]: I understand the reasoning behind splitting socket chat app and standard flask service, but your `tp_chat` package still requires modules from `tp_models`. It is little tangled, maybe reconsider small structure changes eg. `teaching_platform.models` (or different name, for your main flask app logic) , `teaching_platform.chat` (for chat).

File: tp_chat/handler.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import logging
+
+from tp_chat.extensions import socket
+
+from flask_socketio import send
+
+@socket.on("message")
+def handle_message(message):
+    logging.info(f"Recived message: {message}")
+    if message == "User connected.":
```
Comment (cansubmarinesswim) [inline tp_chat/handler.py]: Why include this specific if statement? If its related to `socket.send("User {{username}} connected.");` then it is basically always false still (due to username != "").

File: tp_models/users/controller.py (hunk 1 above)
Comment (Czupryn013) [reply tp_models/users/controller.py]: done

File: tp_chat/handler.py (hunk 3 above)
Comment (Czupryn013) [reply tp_chat/handler.py]: done

File: setup.py (hunk 2 above)
Comment (Czupryn013) [reply setup.py:3]: Done. But imports are a little long. For example in `main.py` in line 11:
`from teaching_platform.models.projects.controller import project_controller_bp`

**Pull request merged**